src/main/java/io/cdap/plugin/hubspot/common/BaseHubspotConfig.java

```diff
@@ -0,0 +1,266 @@
+  @Macro
+  @Nullable
+  public String startDate;
+  @Name(END_DATE)
+  @Description("End date for the analytics report data. YYYYMMDD format.")
+  @Macro
+  @Nullable
+  public String endDate;
+  @Name(REPORT_TYPE)
+  @Description("The analytics report type of content that you want to get data for. Must be one of:\n" +
+    "\n" +
+    "landing-pages - Pull data for landing pages. \n" +
+    "standard-pages - Pull data for website pages. \n" +
+    "blog-posts - Pull data for individual blog posts. \n" +
+    "listing-pages - Pull data for blog listing pages.\n" +
+    "knowledge-articles - Pull data for knowledge base articles.\n" +
+    "\n" +
+    "The analytics report category used to break down the analytics data. Must be one of:\n" +
+    "\n" +
+    "totals - Data will be the totals rolled up from \n" +
+    "sessions - Data broken down by session details \n" +
+    "sources - Data broken down by traffic source\n" +
+    "geolocation - Data broken down by geographic location\n" +
+    "utm-:utm_type - Data broken down by the standard UTM parameters. " +
+    ":utm_type must be one of campaigns, contents, mediums, sources, or terms (i.e. utm-campaigns).\n" +
+    "\n" +
+    "The analytics report type of object that you want the analytics data for. Must be one of:\n" +
+    "\n" +
+    "event-completions - Get data for analytics events. The results are broken down by the event ID." +
+    " You can get the details for the events using this endpoint.\n" +
+    "forms - Get data for your HubSpot forms. The results are broken down by form guids." +
+    " You can get the details for the forms through the Forms API. \n" +
+    "pages - Get data for all URLs with data collected by HubSpot tracking code." +
+    " The results are broken down by URL. \n" +
+    "social-assists - Get data for messages published through the social publishing tools." +
+    " The results are broken down by the broadcastGuid of the messages." +
+    " You can get the details of those messages through the Social Media API.")
+  @Macro
+  @Nullable
+  public String reportType;
+  @Name(TIME_PERIOD)
+  @Description("The time period used to group the data. Must be one of:\n" +
+    "\n" +
+    "total - Data is rolled up to totals covering the specified time.\n" +
+    "daily - Grouped by day\n" +
+    "weekly - Grouped by week\n" +
+    "monthly - Grouped by month\n" +
+    "summarize/daily - Grouped by day, data is rolled up across all breakdowns \n" +
+    "summarize/weekly - Grouped by week, data is rolled up across all breakdowns \n" +
+    "summarize/monthly - Grouped by month, data is rolled up across all breakdowns")
+  @Macro
+  @Nullable
+  public String timePeriod;
+  @Name(APP_ID)
+  @Description("OAuth2 App ID")
+  @Macro
+  public String appId;
+  @Name(API_KEY)
+  @Description("OAuth2 API Key")
+  @Macro
+  public String apiKey;
+  @Name(OBJECT_TYPE)
+  @Description("Select from Contact Lists, Contacts, Email Events, Email Subscription," +
```

Comment:
no need to list here all objects, just replace them with like "select from Hubspot objects/fields. i.e. any what fits by the logic"